src/rgw/rgw_asio_client.cc

```diff
@@ -47,22 +47,22 @@ int ClientIO::init_env(CephContext *cct)
       continue;
     }
 
-    static const std::string_view HTTP_{"HTTP_"};
-
-    char buf[name.size() + HTTP_.size() + 1];
-    auto dest = std::copy(std::begin(HTTP_), std::end(HTTP_), buf);
-    for (auto src = name.begin(); src != name.end(); ++src, ++dest) {
-      if (*src == '-') {
-        *dest = '_';
-      } else if (*src == '_') {
-        *dest = '-';
-      } else {
-        *dest = std::toupper(*src);
-      }
-    }
-    *dest = '\0';
-
-    env.set(buf, std::string(value));
+    static constexpr std::string_view HTTP_{"HTTP_"};
+
+    std::string key{HTTP_};
```

Comment:
It may or may not be worth calling reserve() on the final size (as happens in rgw_common.cc); back_inserter<> can be slightly sluggish if there are repeated reallocations. However, I have not measured and my intuition says it's probably not at all an issue. Maybe worth considering, though.

Reply:
I think we are, unless I'm missing something.